chore: modernize PR template (#92612)
NixOS/nixpkgs: .github/PULL_REQUEST_TEMPLATE.md

###### Things done

<!-- 
It is strongly suggested to use `nipkgs-review`:
- install `nixpkgs-review` for good with `nix-env -f '<nixpkgs>' -iA nixpkgs-review`
- consult usage documentation: https://github.com/Mic92/nixpkgs-review#usage
-->

- [ ] Tested sucessful build of final PR `nixpkgs-review pr <pr-number>`.
      If suceeded, within the resulting `nix-shell`:
   - [ ] Manually tested execution of all binary files (in `./results/bin/`)
   - [ ] Mentioned the manual checks and validations at the end of generated `./report.md`
   - [ ] If ok, posted the results: `nix-shell> nixpkgs-review post-result`

- Platform(s), I built on (reviewers, please complement!):
<!-- more is better, reviewers might complement -->
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions (Ubuntu, Arch Linux, Alpine, etc.).

- [ ] If available: tested via one or more NixOS test(s) `nixpkgs-review pr -p nixosTests.<test> <package> <pr-number>` (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests)

> <!-- Note, that only few tests are available, if you'd want to write your own have a look at: https://github.com/NixOS/nixpkgs/issues/34987 and furthermore read through https://nixos.org/nixos/manual/index.html#sec-nixos-tests. Looking through the source code should be helpful as well. -->

Reviewer: Really, the  [installed tests](https://wiki.gnome.org/Initiatives/GnomeGoals/InstalledTests) are only relevant to fraction of GNOME and GNOME-adjacent software, most of which already has a package so I would not waste characters on promoting it here. Saying that as person interested in these tests.

###### Things done

<!-- 

> Requirements: 

Reviewer: I am not convinced we should make `nixpkgs-preview` a hard requirement at this time. It is [too much memory](https://github.com/Mic92/nixpkgs-review/issues/103) so running it is not an option for some reviewers. (I can only run it if I quit Firefox, or my computer freezes when trying to swap :crying_cat_face:)

- install `nixpkgs-review` for good with `nix-env -f '<nixpkgs>' -iA nixpkgs-review`
- consult usage documentation: https://github.com/Mic92/nixpkgs-review#usage
- setup github api token: https://github.com/Mic92/nixpkgs-review#github-api-token
-->

> - [ ] Tested sucessful build of final PR `GITHUB_TOKEN=<YOUR_TOKEN> nixpkgs-review pr <pr-number>`.

Reviewer: Similarly, it might be more convenient to have `nixpkgs-review` ask for the token and store it.

> - [ ] Tested sucessful build of final PR `GITHUB_TOKEN=<YOUR_TOKEN> nixpkgs-review pr <pr-number>`.

Reviewer: Also not sure it makes sense to mention `pr <pr-number>` in the checklist since when a person is contributing, there is no PR-yet.

For “built on platforms”, I can do that before submitting the PR, but here it forces me to leave the terminal to go edit the pull request after submitting it. I use [`gh`](https://github.com/cli/cli) so my current workflow is running `nix-build` and then checking the checkboxes in the text editor, before submitting.

Author: This looks like a good workflow. However:
- The pullrequesters (now) iare encuraged to complete the same worfklow as the reviewers
- That is, they are encouraged to `nixpkgs-review post-result` this is only available once the pr exists

This can't be addressed in this PR, but it might make sense to support creating a PR with `nixpkgs-review` altogether so that already the commit message can combine:
- PR template for the respective PR type (module update, package update, new, etc) with type-specific check boxes
- concatenated with the result of the nixpkgs-review evaluation

Author: https://github.com/Mic92/nixpkgs-review/issues/115

Reviewer: Well, there is `nixpkgs-review rev HEAD` which can be used by the pullrequester, maybe we could add a comment like ``<!-- you can use `nixpkgs-review rev HEAD` to run the tests before PR is opened -->``

Reviewer: Well, there is `nixpkgs-review rev HEAD` which can be used by the pullrequester, maybe we could add a comment like ``<!-- you can use `nixpkgs-review rev HEAD` to run the tests before PR is opened -->``

      If suceeded, within the resulting `nix-shell`:
   - [ ] Manually tested execution of all binary files (in `./results/bin/`)

>    - [ ] Included manual checks and validations at the end of `editor ./report.md`

Reviewer: Maybe something like

```suggestion
   - [ ] Mentioned the manual checks and validations at the end of generated `./report.md`
```

   - [ ] If ok, posted the results: `nix-shell> nixpkgs-review post-result`

- Platform(s), I built on (reviewers, please complement!):
<!-- more is better, reviewers might complement -->
   - [ ] NixOS
   - [ ] macOS

>    - [ ] other Linux (Ubuntu, Archlinux, Alpine, etc.).

Reviewer: ```suggestion
   - [ ] other Linux distributions (Ubuntu, Arch Linux, Alpine, etc.).
```

but I do not see the need to list examples here.

Author: It came up as a question in the discussion, let's keep it. Eventually we might ask for the distribution,a s well: glibc vs musllib and stuff...

> - [ ] If available: tested via one or more NixOS test(s) `GITHUB_TOKEN=<YOUR_TOKEN> nixpkgs-review pr -p nixosTests.<test> <package> <pr-number>` (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests)

Reviewer: We might also want to mention `passthru.tests`. Or implement https://github.com/Mic92/nixpkgs-review/issues/77 and add the mapping to every package with NixOS test.

Author: Could you make a suggestion until the mentioned PR is merged, or should we just be pationt until then and leave this as it is?

Reviewer: Maybe we could add a comment like “look for `passthru.tests` within the package expression” or “try running `nix-build -A <package>.tests`”. Or just ignore it for now, since ofborg will already run `passthru.tests` so there is not much need to run them locally. I guess that is fine, no need to ask reviewers to do what machines can.

> <!-- Note, that only few tests are available, if you'd want to write your own have a look at: https://github.com/NixOS/nixpkgs/issues/34987 and other tests throughout the source. >

Reviewer: ```suggestion
<!-- Note, that only few tests are available, if you'd want to write your own have a look at: https://github.com/NixOS/nixpkgs/issues/34987 and other tests throughout the source. -->
```

> <!-- Note, that only few tests are available, if you'd want to write your own have a look at: https://github.com/NixOS/nixpkgs/issues/34987 and other tests throughout the source. >

Reviewer: Also, installed tests are only provided by very few projects, so linking to something like https://nixos.org/nixos/manual/index.html#sec-nixos-tests might be more useful to people.

> - [ ] No documentation affected by this change
> 
> - [ ] Or: ensured that relevant documentation is up to date

Reviewer: ```suggestion
- [ ] Ensured that relevant documentation is up to date
```

I think the previous formulation was neater. It was shorter and when there is no relevant documentation, all relevant documentation is trivially up to date.

###### Things done

<!-- 
Requirements: 
- install `nixpkgs-review` for good with `nix-env -f '<nixpkgs>' -iA nixpkgs-review`
- consult usage documentation: https://github.com/Mic92/nixpkgs-review#usage
- setup github api token: https://github.com/Mic92/nixpkgs-review#github-api-token
-->

- [ ] Tested sucessful built of final PR `GITHUB_TOKEN=<YOUR_TOKEN> nixpkgs-review pr <pr-number>`.
      If suceeded, within the resulting `nix-shell`:

>    - [ ] Manually tested execution of all binary files (usually in `./result/bin/`)

Author: Those might be actually in the environment, right?

   - [ ] If ok, posted the results: `nix-shell> nixpkgs-review post-result`

- Tick the the platform(s) at hand that you did built on (more is better, reviewers are encouraged to complement):
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux (Ubuntu, Archlinux, Alpine, etc.). Please specify: ...

> - [ ] If available*: tested via one or more NixOS test(s) `GITHUB_TOKEN=<YOUR_TOKEN> nixpkgs-review pr -p nixosTests.<test> <package> <pr-number>` (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests)

Author: Is this `<test> <package>` or more precisely rather `<package> <package>`? The usage example was ambiguous.

- [ ] No documentation affected by this change
- [ ] Or: Ensured that relevant documentation is up to date

- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

> \* Note, that only few tests are available, if you'd want to write your own have a look at: https://github.com/NixOS/nixpkgs/issues/34987 and other tests throughout the source.

Author: Second voice? or rather a comment?

###### Things done

<!-- 
Requirements: 
- install `nixpkgs-review` for good with `nix-env -f '<nixpkgs>' -iA nixpkgs-review`
- consult usage documentation: https://github.com/Mic92/nixpkgs-review#usage

> - setup github api token: https://github.com/Mic92/nixpkgs-review#github-api-token

Reviewer 2: I think we can skip this step. The github token is only required for advanced features.

Author: I have a split argument about this:
- pull requester will be nudged into graduating into reviewers
- it helps towards a standardized review process if the pull requester already does: `nixpkgs-review post-result`

(and also: don't be too easy on pull requester, to shift the balance away from reviewers)

What do you think about those arguments?

Reviewer: Could we make the request for API token part of first-time running nixpkgs-review? That way we would not have to mention it here.

Author: @jtojnar This is a really nice suggestion! I'll take it out here and open an issue to track it.

-->

- [ ] Tested sucessful built of final PR `GITHUB_TOKEN=<YOUR_TOKEN> nixpkgs-review pr <pr-number>`.
      If suceeded, within the resulting `nix-shell`:
   - [ ] Manually tested execution of all binary files (in `./results/bin/`)

>    - [ ] Include the manual checks and validations you did at the end in `editor ./report.md`

Author: Akin to https://github.com/Mic92/nixpkgs-review/issues/113

   - [ ] If ok, posted the results: `nix-shell> nixpkgs-review post-result`

- Platform(s), I built on (reviewers, please complement!):
<!-- more is better, reviewers might complement -->
   - [ ] NixOS
   - [ ] macOS

>    - [ ] other Linux (Ubuntu, Archlinux, Alpine, etc.). Please specify: ...

Reviewer 2: I don't think it needs to be specified:

```suggestion
   - [ ] other Linux (Ubuntu, Archlinux, Alpine, etc.).
```

Author: However, maybe it's relevant if it's run on glibc or musllibc ?
